app/repositories/schemepreferences/SchemePreferencesRepository.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.schemepreferences
+
+import model.Exceptions.{CannotUpdateSchemePreferences, SchemePreferencesNotFound}
+import model.SelectedSchemes
+import reactivemongo.api.DB
+import reactivemongo.bson.{BSONDocument, BSONObjectID, _}
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait SchemePreferencesRepository {
+  def find(applicationId: String): Future[SelectedSchemes]
+
+  def save(applicationId: String, schemePreferences: SelectedSchemes): Future[Unit]
+}
+
+class SchemePreferencesMongoRepository(implicit mongo: () => DB)
+  extends ReactiveRepository[SelectedSchemes, BSONObjectID]("application", mongo,
+    SelectedSchemes.selectedSchemesFormat, ReactiveMongoFormats.objectIdFormats) with SchemePreferencesRepository {
+  val SchemePreferencesCollection = "scheme-preferences"
+
+  def find(applicationId: String): Future[SelectedSchemes] = {
+    val query = BSONDocument("applicationId" -> applicationId)
+    val projection = BSONDocument(SchemePreferencesCollection -> 1, "_id" -> 0)
+
+    collection.find(query, projection).one[BSONDocument] map {
+      case Some(document) if document.getAs[BSONDocument](SchemePreferencesCollection).isDefined =>
+        document.getAs[SelectedSchemes](SchemePreferencesCollection).get
+      case _ => throw SchemePreferencesNotFound(applicationId)
+    }
+  }
+
+  def save(applicationId: String, schemePreference: SelectedSchemes): Future[Unit] = {
+    val query = BSONDocument("applicationId" -> applicationId)
+    val preferencesBSON = BSONDocument("$set" -> BSONDocument(
+      "scheme-preferences" -> schemePreference,
+      "progress-status." + SchemePreferencesCollection -> true
```

Comment:
Should there be a dedicated Progress Status in the progress statuses class for this?